lib/app.js

```diff
@@ -83,39 +96,50 @@ class Container extends Component {
       // https://github.com/gaearon/react-hot-loader/issues/442
       return (
         <AppContainer warnings={false} errorReporter={ErrorDebug}>
-          <Component {...props} url={url} />
+          {children}
         </AppContainer>
       )
     }
   }
 }
 
-function createUrl (router) {
+const warnUrl = execOnce(() => warn(`Warning: the 'url' property is deprecated. https://err.sh/next.js/url-deprecated`))
+
+export function createUrl (router) {
   return {
-    query: router.query,
-    pathname: router.pathname,
-    asPath: router.asPath,
+    get query () {
+      warnUrl()
+      return router.query
+    },
+    get pathname () {
+      warnUrl()
+      return router.pathname
+    },
+    get asPath () {
+      warnUrl()
+      return router.asPath
+    },
     back: () => {
-      warn(`Warning: 'url.back()' is deprecated. Use "window.history.back()"`)
+      warn(`Warning: 'url.back()' is deprecated. Use "window.history.back()" https://err.sh/next.js/url-deprecated`)
```

Comment:
Don't we use something like `warnOnce` or something.

Reply:
I believe this was deprecated in v2 or v3. That's what I didn't want to touch it too much, just adding the url to the message. In theory we could use the same message though. Will update it.